### ISSUE TITLE
[menu] global context properties

### ISSUE THREAD

**Issue by Guria** [bem/bem-components#1516]
`_firstItem` and `_checkedItems` properties [set as global](https://github.com/bem/bem-components/blob/v2.1.1/common.blocks/menu/menu.bemhtml#L36-L37) in context.

``` js
          { block:'menu', content:[{block:'menu-item', content:'asdasd'}]},
          { block: 'debug'},
```

``` js
block('debug').content()(function(){
    return JSON.stringify(this._firstItem);
});
```

``` html
<div class="menu menu__control i-bem" data-bem="{&quot;menu&quot;:{}}" role="menu" tabindex="0">
  <div class="menu-item i-bem" data-bem="{&quot;menu-item&quot;:{}}" role="menuitem">asdasd</div></div>
<div class="debug">{"block":"menu-item","content":"asdasd","mods":{}}</div>
```

**Comment by narqo** [comment]
Nice catch, thanks! We will fix it in the nearest future

**Comment by narqo** [comment]
Actually, it shouldn't hurt any "real life" usage (as these flags will be reassigned for each menu instance). 

But anyway, could you make a separate PR for the issue, so we will merge it?

**Comment by Guria** [comment]
Okay

**Comment by qfox** [comment]
:+1: for that fix but looks like it's breaking change.

**Comment by Guria** [comment]
@zxqfox #1523 just bug fix, #1520 more serious change but I don't see what it can brake
